templates/registration/registration_form.html

```diff
@@ -24,7 +24,7 @@ <h2 class="align-right">{% trans "Create a new account" %}</h2>
                    class="form-control"
                    placeholder="{% trans "Email" %}"
                    name="email"
-                   value="{{ form.cleaned_data.email }}">
+                   value="{{ form.cleaned_data.email|default:form.email.value }}">
```

Comment:
```suggestion
                   value="{{ form.cleaned_data.email|default:form.email.value|default:"" }}">
```
Avoid "None" in form field, hiding "placeholder" value and confusing users.